azurelinuxagent/common/event.py

```diff
@@ -242,57 +242,55 @@ def is_period_elapsed(self, delta, h):
         return h not in self.periodic_events or \
             (self.periodic_events[h] + delta) <= datetime.now()
 
-    def add_periodic(self,
-                     delta, name, op=WALAEventOperation.Unknown, is_success=True, duration=0,
-                     version=CURRENT_VERSION, message="", evt_type="",
-                     is_internal=False, log_event=True, force=False):
-
+    def add_periodic(self, delta, name, op=WALAEventOperation.Unknown, is_success=True, duration=0,
+                     version=str(CURRENT_VERSION), message="", evt_type="", is_internal=False, log_event=True,
+                     force=False, event_time_opcode_name=None, event_pid=0, event_tid=0, task_name="", keyword_name=""):
```

Comment:
Instead of passing `event_time_opcode_name=None` this as a parameter? Why not set it in the `_add_default_parameters_to_event` function? 
I'm trying to think of scenarios where we would set it to something other than the time the event occurred. Is there some scenario that I'm missing?

Reply:
I am already setting the event_time_opcode_name in `_add_default_parameters_to_event` [(link here)](https://github.com/Azure/WALinuxAgent/pull/1663/files#diff-58adf10cb50d782909fbf94ac6c08bf5R359) if it is not passed. Just exposing it to the caller if the caller wants to pass it.

```
event.parameters.append(TelemetryEventParam('OpcodeName', event_time_opcode_name if event_time_opcode_name else datetime.utcnow().__str__()))
```

Reply:
ohh sorry missed it, thanks for pointing it out!

Reply:
> Just exposing it to the caller if the caller wants to pass it.

Do you anticipate any scenario where this would be needed?

Reply:
(similar for pid & tid)